src/storage.cc

```diff
@@ -0,0 +1,386 @@
+                                       << " kg of " << in_commod
+                                       << " recipe: " << in_recipe << ".";
+  }
+
+  LOG(cyclus::LEV_INFO3, "ComCnv") << "}";
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::Tock() {
+  LOG(cyclus::LEV_INFO3, "ComCnv") << prototype() << " is tocking {";
+
+  BeginProcessing_(); // place unprocessed inventory into processing
+
+  if( ready() >= 0 || process_time_() == 0 ) {
+    Convert_(capacity_()); // place processing into stocks
+  }
+
+  LOG(cyclus::LEV_INFO3, "ComCnv") << "}";
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
+Storage::GetMatlRequests() {
+  using cyclus::CapacityConstraint;
+  using cyclus::Material;
+  using cyclus::RequestPortfolio;
+  using cyclus::Request;
+
+  std::set<RequestPortfolio<Material>::Ptr> ports;
+
+  RequestPortfolio<Material>::Ptr port(new RequestPortfolio<Material>());
+  Material::Ptr mat = Request_();
+  double amt = mat->quantity();
+
+  if (amt > cyclus::eps()) {
+    //CapacityConstraint<Material> cc(amt);
+    //port->AddConstraint(cc);
+
+    port->AddRequest(mat, this, in_commod_());
+
+    ports.insert(port);
+  }
+
+  return ports;
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::AcceptMatlTrades(
+  const std::vector< std::pair<cyclus::Trade<cyclus::Material>,
+  cyclus::Material::Ptr> >& responses) {
+  std::vector< std::pair<cyclus::Trade<cyclus::Material>,
+      cyclus::Material::Ptr> >::const_iterator it;
+  for (it = responses.begin(); it != responses.end(); ++it) {
+    AddMat_(it->second);
+  }
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr>
+Storage::GetMatlBids(cyclus::CommodMap<cyclus::Material>::type&
+                          commod_requests) {
+  using cyclus::BidPortfolio;
+  using cyclus::Material;
+
+  std::set<BidPortfolio<Material>::Ptr> ports;
+
+  std::set<std::string>::const_iterator it;
+  BidPortfolio<Material>::Ptr port = GetBids_(commod_requests,
+                                              out_commod,
+                                              &stocks);
+  if (!port->bids().empty()) {
+    ports.insert(port);
+  }
+
+  return ports;
+
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::GetMatlTrades(
+  const std::vector< cyclus::Trade<cyclus::Material> >& trades,
+  std::vector<std::pair<cyclus::Trade<cyclus::Material>,
+  cyclus::Material::Ptr> >& responses) {
+  using cyclus::Material;
+  using cyclus::Trade;
+
+  // for each trade, respond
+  std::vector< Trade<Material> >::const_iterator it;
+  for (it = trades.begin(); it != trades.end(); ++it) {
+    std::string commodity = it->request->commodity();
+    //double qty = it->amt;
```

Comment:
We should determine whether this or the line below it is the desired behavior. I suspect the line below is the desired implementation, but we should make sure. If that turns out to be the case, then the commented line should be deleted.